pyface/resource/resource_manager.py

```diff
@@ -127,7 +127,7 @@ def _locate_image(self, image_name, resource_path, size):
                             return ImageReference(
                                 self.resource_factory, data=data
                             )
-                        except IOError:
+                        except (IOError, TypeError):
```

Comment:
I am not sure about capturing `TypeError` here. It often indicates a programming error, and it is a programming error to try to pass `"__main__"` to `files`. It would look more correct if the scenario is handled earlier. Plus capturing `TypeError` will also capture other programming errors we want to know about and fix.